src/pretix/static/pretixbase/js/addressform.js

```diff
@@ -45,80 +118,44 @@ $(function () {
             if (dependents.transmission_type) {
                 url.searchParams.append("transmission_type_required", !dependents.transmission_type.find("option[value='-']").length);
             }
-            xhr = $.getJSON(url, function (data) {
-                var selected_state = dependents.state.prop("data-selected-value");
-                if (selected_state) dependents.state.prop("data-selected-value", "");
-                dependents.state.find("option:not([value=''])").remove();
-                $.each(data.data, function (k, s) {
-                    var o = $("<option>").attr("value", s.code).text(s.name);
-                    if (selected_state === s.code) o.prop("selected", true);
-                    dependents.state.append(o);
-                });
 
-                if (dependents.transmission_type) {
-                    var selected_transmission_type = dependents.transmission_type.prop("data-selected-value");
-                    if (selected_transmission_type) dependents.transmission_type.prop("data-selected-value", "");
-                    dependents.transmission_type.find("option:not([value='']):not([value='-'])").remove();
-
-                    if (!data.transmission_type.visible) {
-                        selected_transmission_type = "email";
+            function load() {
+                xhr = $.getJSON(url, function (data) {
+                    responseCache[url] = data;
+                    update_form(data);
+                }).always(function() {
+                    loader.hide();
+                }).fail(function(){
+                    // In case of errors, show everything and require nothing, we can still handle errors in backend
```

Comment:
```suggestion
                    if (responseCache[url] === "LOADING") delete responseCache[url];

                    // In case of errors, show everything and require nothing, we can still handle errors in backend
```